dmoj/problem.py

```diff
@@ -396,3 +392,25 @@ def __getstate__(self):
 
     def __setstate__(self, state):
         self.__dict__.update(state)
+
+
+class BatchedTestCase(AbstractTestCase):
+    def __init__(self, position, batch, config, problem):
+        super().__init__(position, batch, config, problem)
+
+
+class StandaloneTestCase(AbstractTestCase, TopLevelCase):
+    def __init__(self, top_level_position, position, config, problem):
+        AbstractTestCase.__init__(self, position, 0, config, problem)
```

Comment:
Do things break if we do this?

```suggestion
        AbstractTestCase.__init__(self, position, None, config, problem)
```

Reply:
I didn't test it. I'm not even sure we need this argument on `AbstractTestCase`. Admittedly, I just mirrored the old behaviour.

Reply:
Yeah, it'd be nice to either drop the batch parameter from the `AbstractTestCase` class, or ensure that `None` is valid here. Using `0` to mean "no batch" is kind of sketchy.

Reply:
Oh I remember. The batch gets passed to checkers. https://github.com/DMOJ/judge-server/blob/master/dmoj/graders/standard.py#L61

So test cases have to have a batch field. Do you want to make it `None`? That seems better, but will it break anything?

Reply:
Let's just make it `None` and roll back if prod breaks ¯\_(ツ)_/¯